src/relay/pass/type_infer.cc

```diff
@@ -346,7 +391,20 @@ class TypeInferencer::Resolver : public ExprMutator {
   }
 
   Expr VisitExpr_(const CallNode* op) final {
-    return AttachCheckedType(op);
+    auto call = GetRef<Call>(op);
+    auto it = tmap_.find(call);
+    if (it != tmap_.end()) {
+      Call new_op = Downcast<Call>(AttachCheckedType(op));
```

Comment:
Maybe we can change AttachCheckedType -> AttachTypeInfo, and directly attach type_args when it is defined